src/moonlink_backend/src/error.rs

```diff
@@ -46,6 +46,16 @@ pub enum Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
+impl Error {
+    #[track_caller]
+    pub fn invalid_argument(mode: &str) -> Self {
```

Comment:
invalid argument is a very general purpose error, which shouldn't be limited to table optimization function